test/integration/navsat.cc

```diff
@@ -0,0 +1,332 @@
+  sensorNoise->SetVelocity(velocity);
+  EXPECT_EQ(velocity, sensor->Velocity());
+  EXPECT_EQ(velocity, sensorNoise->Velocity());
+
+  // verify msg received on the topic
+  WaitForMessageTestHelper<msgs::NavSat> msgHelper(topic);
+  sensor->Update(std::chrono::steady_clock::duration(1s));
+  EXPECT_TRUE(msgHelper.WaitForMessage()) << msgHelper;
+  auto msg = msgHelper.Message();
+  EXPECT_EQ(1, msg.header().stamp().sec());
+  EXPECT_EQ(0, msg.header().stamp().nsec());
+  EXPECT_DOUBLE_EQ(lat.Degree(), msg.latitude_deg());
+  EXPECT_DOUBLE_EQ(lon.Degree(), msg.longitude_deg());
+  EXPECT_DOUBLE_EQ(altitude, msg.altitude());
+  EXPECT_DOUBLE_EQ(velocity.X(), msg.velocity_east());
+  EXPECT_DOUBLE_EQ(velocity.Y(), msg.velocity_north());
+  EXPECT_DOUBLE_EQ(velocity.Z(), msg.velocity_up());
+
+  // verify msg with noise received on the topic
+  WaitForMessageTestHelper<msgs::NavSat> msgHelperNoise(topicNoise);
+  sensorNoise->Update(std::chrono::steady_clock::duration(1s));
+  EXPECT_TRUE(msgHelperNoise.WaitForMessage()) << msgHelperNoise;
+  auto msgNoise = msgHelperNoise.Message();
+  EXPECT_EQ(1, msgNoise.header().stamp().sec());
+  EXPECT_EQ(0, msgNoise.header().stamp().nsec());
+  EXPECT_FALSE(math::equal(lat.Degree(), msgNoise.latitude_deg()));
+  EXPECT_FALSE(math::equal(lon.Degree(), msgNoise.longitude_deg()));
+  EXPECT_FALSE(math::equal(altitude, msgNoise.altitude()));
+  EXPECT_FALSE(math::equal(velocity.X(), msgNoise.velocity_east()));
+  EXPECT_FALSE(math::equal(velocity.Y(), msgNoise.velocity_north()));
+  EXPECT_FALSE(math::equal(velocity.Z(), msgNoise.velocity_up()));
+}
+
+/////////////////////////////////////////////////
+TEST_F(NavSatSensorTest, Topic)
+{
+  const std::string name = "TestNavSat";
+  const double updateRate = 30;
+  const bool alwaysOn = 1;
+  const bool visualize = 1;
+  auto sensorPose = math::Pose3d();
+
+  // Factory
+  SensorFactory factory;
+
+  // Default topic
+  {
+    const std::string topic;
+    auto navsatSdf = NavSatToSdf(name, sensorPose, updateRate, topic,
+        alwaysOn, visualize);
+
+    auto navsat = factory.CreateSensor<NavSatSensor>(navsatSdf);
+    ASSERT_NE(nullptr, navsat);
+
+    EXPECT_EQ("/navsat", navsat->Topic());
+  }
+
+  // Convert to valid topic
+  {
+    const std::string topic = "/topic with spaces/@~characters//";
+    auto navsatSdf = NavSatToSdf(name, sensorPose, updateRate, topic,
+        alwaysOn, visualize);
+
+    auto navsat = factory.CreateSensor<NavSatSensor>(navsatSdf);
+    ASSERT_NE(nullptr, navsat);
+
+    EXPECT_EQ("/topic_with_spaces/characters", navsat->Topic());
+  }
+
+  // Invalid topic
+  {
+    const std::string topic = "@@@";
+    auto navsatSdf = NavSatToSdf(name, sensorPose, updateRate, topic,
+        alwaysOn, visualize);
+
+    auto sensor = factory.CreateSensor<
+        NavSatSensor>(navsatSdf);
+    ASSERT_EQ(nullptr, sensor);
+  }
+}
+
+int main(int argc, char **argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
```

Comment:
general question/note about the tests: I didn't see anything that tested the noise behavior of a navsat sensor. Is that something that should be tested, or is this considered "already tested" by https://github.com/ignitionrobotics/ign-sensors/blob/ign-sensors6/src/Noise_TEST.cc?

Reply:
We check that the values have been modified on lines 272~277. That's enough to catch a situation where we forgot to apply noise, for example. It would not catch it if we mixed up the noises and applied them to the wrong values though :thinking: 

I don't think it's necessary to check that the noise is behaving as expected here, I think that should be covered by `Noise_TEST`.